app/src/org/commcare/models/database/user/models/CommCareEntityStorageCache.java

```diff
@@ -72,18 +73,12 @@ public static void createIndexes(SQLiteDatabase db) {
         db.execSQL(DatabaseIndexingUtils.indexOnTableCommand("NAME_ENTITY_KEY", TABLE_NAME, COL_CACHE_NAME + ", " + COL_ENTITY_KEY + ", " + COL_CACHE_KEY));
     }
 
-    public boolean lockCache() {
+    public Closeable lockCache() {
         //Get a db handle so we can get an outer lock
-        SQLiteDatabase db;
-        try {
-            db = CommCareApplication.instance().getUserDbHandle();
-        } catch (SessionUnavailableException e) {
-            return false;
-        }
-
+        SQLiteDatabase db = CommCareApplication.instance().getUserDbHandle();
         //get the db lock
         db.beginTransaction();
-        return true;
+        return db;
```

Comment:
Doing this will result in the DB connection being closed as well (instead of just ending the transaction). Is that what we want?

Reply:
Ahh right! I don't think we want to `close` the db. Should we just return a dummy closeable in that case ?

Reply:
Yes, we can create a new class that will just end the transaction in the close method.